### PR TITLE
Armstrong_Number.dart

### PR DESCRIPTION
**Pull request by pooja-git11**
Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
First of all complete your PR template.

File: Armstrong_Number/armstrong_number.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    var num=int.parse(input);
+    
+    bool check=is_armstrong(num,len);
+    
+    if(check==true)
+        print("YES");
+    else
+        print("NO");
+    
+}
```
Comment (raghav-dalmia) [inline Armstrong_Number/armstrong_number.dart]: add sample i/o

File: Armstrong_Number/armstrong_number.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    */
+
+    while(n1>0)
+    {
+        rem=n1%10;
+        sum=sum+pow(rem,l);
+        n1=(n1/10).toInt();
+        
+    }
+    if(sum==n)
```
Comment (raghav-dalmia) [inline Armstrong_Number/armstrong_number.dart]: add space around operators(<>,==;!)

File: Armstrong_Number/armstrong_number.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import 'dart:math';
+
+
+bool is_armstrong(int n,int l)
+{
+    int n1,rem,sum=0; 
+    n1=n;
+    
+    /*
+        An Armstrong number is a number that is equal to the sum of digits raise to the power total number of digits in the number
```
Comment (raghav-dalmia) [inline Armstrong_Number/armstrong_number.dart]: move this line in the beginning of the code

File: Armstrong_Number/armstrong_number.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    */
+
+    while(n1>0)
+    {
+        rem=n1%10;
+        sum=sum+pow(rem,l);
+        n1=(n1/10).toInt();
+
```
Comment (raghav-dalmia) [inline Armstrong_Number/armstrong_number.dart]: remove extra blank line

File: Armstrong_Number/armstrong_number.dart [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        sum=sum+pow(rem,l);
+        n1=(n1/10).toInt();
+        
+    }
+    if(sum==n)
+        return true;
+    
+    return false;
+}
+main(){
```
Comment (raghav-dalmia) [inline Armstrong_Number/armstrong_number.dart]: leave a blank line before each function.

**Comment by asha15** [comment]
why did you create two PR for the same issue ?

**Comment by pooja-git11** [comment]
@asha15 actually i am new to github , so i am unable to understand which two PR you are talking about. here i first create PR for armstrong number and then i made changes to it .
so can  you please tell which PR i have to close.

**Comment by MastersAbh** [comment]
Hi @asha15 , was there a dupicate PR for the issue, If this PR stands invalid, can you please close it ?

**Comment by asha15** [comment]
Sorry for the confusion. we can go ahead with the PR

**Comment by asha15** [comment]
could you change the file name to Armstrong_number.dart? then  we are good to merge

File: Armstrong_number.dart [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (somya-kapoor) [inline Armstrong_number.dart:1]: change the file name to  Armstrong_Number/Armstrong_Number.dart

File: Armstrong_number.dart [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+bool is_armstrong(int n,int l)
+{
+    int n1, rem, sum = 0; 
+    n1 = n;
+   
+    while(n1 > 0)
+    {
+        rem = n1 % 10;
+        sum = sum + pow(rem,l);
```
Comment (somya-kapoor) [inline Armstrong_number.dart:17]: add space after ,

File: Armstrong_number.dart [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+main()
+{
+    print("Enter a number  ");
+    var input = stdin.readLineSync();
+    
+    int len = input.length;              //to clalculate the length of string
+    int num = int.parse(input);         //conveting string into integer    
+    
+    bool check = is_armstrong(num,len);
```
Comment (somya-kapoor) [inline Armstrong_number.dart:35]: add space after ,

**Comment by somya-kapoor** [comment]
Please squash the commits

File: Armstrong_number.dart (hunk 6 above)
Comment (somya-kapoor) [reply Armstrong_number.dart]: change the file name to Armstrong_Number/Armstrong_Number.dart

**Comment by somya-kapoor** [comment]
change the file name to Armstrong_Number/Armstrong_Number.dart

**Comment by somya-kapoor** [comment]
Please squash the commits

**Comment by pooja-git11** [comment]
@somya-kapoor  i don't know how to squash the commit , if you can help me any way then you are welcome

**Pull request closed without merging**